src/NUnitEngine/nunit.engine.core/RuntimeFramework.cs

```diff
@@ -444,6 +466,23 @@ public bool Supports(RuntimeFramework target)
                 && this.FrameworkVersion.Minor >= target.FrameworkVersion.Minor;
         }
 
+        private bool Supports(RuntimeType targetRuntime)
+        {
+            if (this.Runtime == targetRuntime)
+                return true;
+
+            if (this.Runtime == RuntimeType.Any || targetRuntime == RuntimeType.Any)
+                return true;
+
+            if (this.Runtime == RuntimeType.Net && targetRuntime == RuntimeType.Mono)
+                return true;
+
+            if (this.Runtime == RuntimeType.Mono && targetRuntime == RuntimeType.Net)
+                return true;
+
+            return false;
```

Comment:
Good move expanding this clause to it's own method. I'm struggling to follow the logic here, but is this a slight behaviour change? What happens now if I use the `--framework=mono` flag when running the console on Windows .NET Framework, would my tests now end up running on netfx instead?

Reply:
I could be wrong, but I don't think it works currently and I don't think it will start working after this either. :confused: 

As I mentioned, I think we have a load of  work to do with RuntimeFramework, which I don't want to do in this PR. One of the big parts of it is to figure out how we handle mono which is more like an alternate implementation of the .NET Framework than a different runtime.

BTW, this is a general method and doesn't specifically relate to the command-line option... i.e. targetFramework may come from the `TargetFrameoworkAttribute`, which never mentions Mono.

Reply:
Definitely agree about refactoring, this area of the code is a nightmare. #702 and #406 track that - one more issue I've never got round to. 😞 

Happy to go with this if you're confident we're not breaking any existing scenarios. I just noticed the logic change, and didn't want us to inadvertantly break anything which is working currently.